netty/src/main/java/io/grpc/netty/NettyChannelBuilder.java

```diff
@@ -406,13 +437,46 @@ protected ClientTransportFactory buildTransportFactory() {
       }
       negotiator = createProtocolNegotiatorByType(negotiationType, localSslContext);
     }
+
+    // TODO(jihuncho) throw exception if not groupOrChannelProvided after 1.22.0
+    ObjectPool<? extends EventLoopGroup> resolvedEventLoopGroupPool = eventLoopGroupPool;
+    ChannelFactory<? extends Channel> resolvedChannelFactory = channelFactory;
+    if (shouldFallBackToNio()) {
+      logger.log(
+          Level.WARNING,
+          "Both EventLoopGroup and ChannelType should be provided, otherwise client may not start. "
```

Comment:
"Both EventLoopGroup and ChannelType should be provided or neither should be, otherwise the client..."

Ditto for server.

Reply:
thanks, done.